### PR TITLE
Release script with pulpcore requirement

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

File: .travis/release.py [hunk 1]
```diff
@@ -136,21 +137,30 @@ def validate_redmine_data(redmine_query_url, redmine_issues):
 git.checkout("HEAD", b=f"release_{release_version}")
 
 # First commit: changelog
+if "pulpcore" not in release_path:
+    pulpcore_requirement = f"pulpcore>={lower_pulpcore_version},<{upper_pulpcore_version}"
```
Comment (quba42) [inline .travis/release.py:141]: From a ReStructuredText formatting point of view, I would prefer to place this string in double backticks. (Not sure if backticks need to be escaped somehow in the context of the `sed()` function.)

See https://github.com/pulp/pulp_deb/pull/219#discussion_r496774407 for an example of the end result I am aming for.

File: .travis/release.py [hunk 2]
```diff
@@ -136,21 +137,30 @@ def validate_redmine_data(redmine_query_url, redmine_issues):
...
-with open(f"{plugin_path}/requirements.txt", "wt") as setup_file:
-    for line in setup_lines:
-        if "pulpcore" in line and "pulpcore" not in release_path:
-            line = f"pulpcore>={lower_pulpcore_version},<{upper_pulpcore_version}\n"
-
-        setup_file.write(line)
+if "pulpcore" not in release_path:
+    sed(
+        "-i",
+        f"s/pulpcore.*/pulpcore>={lower_pulpcore_version},<{upper_pulpcore_version}/",
```
Comment (quba42) [inline .travis/release.py:161]: Why can't this reuse the `pulpcore_requirement` variable from line 141?

**Review by quba42 — COMMENTED** [review]
I can confirm that the script (in its current form) works for Y releases without any issues/changelog entries.

According to my local test, it does not yet add the same string for Y releases that do have issues (and hence changelog entries).

I may have another look at this soon. :wink:

File: .travis/release.py (hunk 2 above)
Comment (fao89) [reply .travis/release.py:161]: actually, I forgot to add it here

File: .travis/release.py (hunk 1 above)
Comment (fao89) [reply .travis/release.py:141]: I've addressed the requested changes and tested it:
https://github.com/pulp/pulp_deb/compare/master...fao89:release_2.7.0

**Pull request closed without merging**